tsl/src/chunk.c

```diff
@@ -200,7 +197,7 @@ chunk_invoke_drop_chunks(Oid relid, Datum older_than, Datum older_than_type)
 	int i, num_results = 0;
 	SetExprState *state;
 	Oid restype;
-	Const *argarr[DROP_CHUNKS_NARGS] = {
+	Const *argarr[] = {
```

Comment:
Could you keep this as DROP_CHUNKS_NARGS? Using  DROP_CHUNKS_NARGS is explicit, the correct number of elts are initialized and additionally is self documenting as it doesn't need looking through the code to find out what is going on.

Reply:
I disagree with that. The number of args should be explicitly specified when creating the array. It is a programming error if the number of args don't match the array specification. It it actually easier to find the programming error when you explicitly specify the number of args and see a mismatch. Are you saying that DROP_CHUNKS_NARGS is incorrect and needs modification?

Reply:
It was incorrect, yes. The point is that we want to either capture problems at compile-time, or avoid the problem if we cannot. Could you explain why you think it is important to both see the number of arguments explicitly in the array declaration as well as in the array initializer?

Reply:
Never mind, I restored the constant names.